### ISSUE TITLE
MX2-8X8-HDMI20 doesnt work

### ISSUE THREAD

**Issue by wvanmaris** [bitfocus/companion-module-lightware-lw3#6]
Using LW3 instance, no in- or outputs are appearing, only ??

Using LW2 the matrix was controllable. Even though the MX2 is mentioned as tested.

**Comment by wvanmaris** [comment]
Hi Dorian,

while I do get to see the input names, the connection blinks every few 
seconds, loosing the input names. Presets are also all blank 
(undefined).

As far as I can see nothing to be set on the Lightware's side.  I can 
disable LW2 protocol in ethernetsettings. And I can enable LW3 in RS232 
settings, but both make no difference.


------ Oorspronkelijke bericht ------
Van "Dorian Meid" ***@***.***>
Aan "bitfocus/companion-module-lightware-lw3" 
***@***.***>
CC "wvanmaris" ***@***.***>; "Author" 
***@***.***>
Datum 17-3-2023 16:08:33
Onderwerp Re: [bitfocus/companion-module-lightware-lw3] MX2-8X8-HDMI20 
doesnt work (Issue #6)

**Comment by JeffreyDavidsz** [comment]
@dnmeid `GET /MEDIA/PRESET/*.Name` does not seem to get any results and ends up in errors processing. Could not find the command in documentation. Working to get everything refactored for v3. Was this a special command for one device?

**Comment by JeffreyDavidsz** [comment]
Getting names in Companion v3 now, able to switch sources. Working on feedback and variable

**Comment by dnmeid** [comment]
The module now has been updated for v3 and has a few improvements.
Can you please retry and report here if there are still some problems with MX2.

**Comment by johnny4x4** [comment]
I am having similar issues with getting an MX2-48x48-HDMI20-A-R to connect using companion v3. Looking at the companion debug, there is a very long error being recorded every time it attempts to connect.
It was originally working- albeit without input/output names- on Companion v2.4.2 and LW3 v1.0.3

**Comment by dnmeid** [comment]
I'd be happy to debug this module for MX2, but it is nearly impossible without connection to a real device. If someone wants to provide a remote connection, feel free to tell me.

**Comment by johnny4x4** [comment]
After making the javascript readable, I found the error occurring on line 614. The error is basically that "e is null" and therefore not iterable. I commented out that section of code and was able to get it working. I have almost zero experience with javascript so I dont really know what negative effects will result from it being removed. I'll just post the section of code that I removed and someone much smarter than me can hopefully create an actual solution:

this.sendCommand('GET /MEDIA/PRESET/*.Name', e => {
                  let t = e.split(/\r\n/)
                  ;(this.CHOICES_PRESETS = t
                    .filter(
                      e =>
                        void 0 !== e.match(/\/MEDIA\/PRESET\/(.+?)\.Name=(.+)$/)
                    )
                    .map(e => {
                      let [t, n, s] = e.match(
                        /\/MEDIA\/PRESET\/(.+?)\.Name=(.+)$/
                      )
                      return { id: n, label: s }
                    })),
                    this.initActions()
                }),

**Comment by JeffreyDavidsz** [comment]
@johnny4x4 just to be sure;
`this.sendCommand('GET /MEDIA/PRESET/*.Name', e => {`
is in the current version;
`this.sendCommand('GET /MEDIA/PRESET/*.Name', (result) => {`
notice the result. Are you using the latest version?

**Comment by dnmeid** [comment]
the 'e' instead of 'result' is from the webpack module minification. Don't worry about it.
Strange thing is that this code actually should not be a problem for MX2.
It reads the available presets from the matrix. MX2 is compatible to the LW3 protocol and this is a valid LW3 get command. Even if the MX2 wouldn't support presets, which it does, it should not respond with `null` but with an error message. The response to any LW3 command should always be a string with either some useful information or an error message.
In the following lines I parse the response for the preset names. If there are no valid preset names or I got an error, it simply does not add the names to the preset choices and skips to the next function.

So I made an update which handles an invalid response here, but actually I'd like to make an update which also can handle presets for the MX2. Please check with the attached packaged module if it works. I made it fail more gracefully at this code and also added some more log messages, so please check the log. Please report back if you have a result.

[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11689310/pkg.tgz)

**Comment by dnmeid** [comment]
I checked again the documentation for the MX2 and as you say there is no command GET /MEDIA/PRESET/*.Name in MX2.
The module should not have crashed by that actually.
I made another version, which first checkes GET /PRESETS/AVC/*.Name and only checks /MEDIA/PRESET/*.Name if that fails.
This is the same command as for the general type of LW3 and it is documented in the MX2 manual.
Maybe you have more luck with this command. I don't know where /MEDIA/PRESET/*.Name came from but it is there since preset recalling had been introduced and I guess it is there for a reason and there is some Lightware matrix which uses that path.
I again attach a package (same name and version, so please tell me carefully which package you are refering to)

[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11689827/pkg.tgz)

**Comment by johnny4x4** [comment]
Thank you! Let me fire up the system and I will see what happens when I plug in your update.

**Comment by dnmeid** [comment]
@johnny4x4 You know how to deal with pkg.tgz?

**Comment by johnny4x4** [comment]
I was able to extract it, I assume all I have to do is drop into the /bundled-modules/lightware-lw3/ folder.

**Comment by dnmeid** [comment]
Depending on your installation yes. Actually better way is to put it in the module-local-dev folder if you have a development environment. If you are running a binary, you have to designate a local folder to be the development modules folder and put it there.
In both cases the development module will override the bundled module without touching it. You can see the override in the log.

**Comment by dnmeid** [comment]
Ok, try this one: 
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11690846/pkg.tgz)

**Comment by johnny4x4** [comment]
Lets see if I did this correctly, is this what you are looking for?:

6B86#OPEN /MEDIA/PRESET
{6B85
n- /MEDIA/PRESET/Horizontal
}
6B87#GET /MEDIA/PRESET
{6B86
o- /MEDIA/PRESET
}
6B88#GET /MEDIA/PRESET.*
{6B87
n- /MEDIA/PRESET/Horizontal
}

**Comment by dnmeid** [comment]
Yes that looks good. So for the `GET /MEDIA/PRESET/.*` is there more results than `-n /MEDIA/PRESET/Horizontal`?
All the lines starting with -n are the available presets.
Later there shoud be a command like `GET /MEDIA/PRESET/Horizontal.*` and some responses starting with 'pw'. Can you please give me them too.

**Comment by johnny4x4** [comment]
Yes, I believe there is a Preset called "horizontal". I've never used the built in presets though because I have been using companion for presets. Here is the rest of the response:

{6B88
pw /MEDIA/PRESET.Button1AssignedPreset=
pw /MEDIA/PRESET.Button2AssignedPreset=
pw /MEDIA/PRESET.Button3AssignedPreset=
pw /MEDIA/PRESET.Button4AssignedPreset=
pw /MEDIA/PRESET.Button5AssignedPreset=
pw /MEDIA/PRESET.Button6AssignedPreset=
pw /MEDIA/PRESET.Button7AssignedPreset=
pw /MEDIA/PRESET.Button8AssignedPreset=
pw /MEDIA/PRESET.Button9AssignedPreset=
pw /MEDIA/PRESET.Button10AssignedPreset=
pw /MEDIA/PRESET.Button11AssignedPreset=
pw /MEDIA/PRESET.Button12AssignedPreset=
pw /MEDIA/PRESET.Button13AssignedPreset=
pw /MEDIA/PRESET.Button14AssignedPreset=
pw /MEDIA/PRESET.Button15AssignedPreset=
pw /MEDIA/PRESET.Button16AssignedPreset=
pw /MEDIA/PRESET.Button17AssignedPreset=
pw /MEDIA/PRESET.Button18AssignedPreset=
pw /MEDIA/PRESET.Button19AssignedPreset=
pw /MEDIA/PRESET.Button20AssignedPreset=
pw /MEDIA/PRESET.Button21AssignedPreset=
pw /MEDIA/PRESET.Button22AssignedPreset=
pw /MEDIA/PRESET.Button23AssignedPreset=
pw /MEDIA/PRESET.Button24AssignedPreset=
pw /MEDIA/PRESET.Button25AssignedPreset=
pw /MEDIA/PRESET.Button26AssignedPreset=
pw /MEDIA/PRESET.Button27AssignedPreset=
pw /MEDIA/PRESET.Button28AssignedPreset=
pw /MEDIA/PRESET.Button29AssignedPreset=
pw /MEDIA/PRESET.Button30AssignedPreset=
pw /MEDIA/PRESET.Button31AssignedPreset=
pw /MEDIA/PRESET.Button32AssignedPreset=
pw /MEDIA/PRESET.Button33AssignedPreset=
pw /MEDIA/PRESET.Button34AssignedPreset=
pw /MEDIA/PRESET.Button35AssignedPreset=
pw /MEDIA/PRESET.Button36AssignedPreset=
pw /MEDIA/PRESET.Button37AssignedPreset=
pw /MEDIA/PRESET.Button38AssignedPreset=
pw /MEDIA/PRESET.Button39AssignedPreset=
pw /MEDIA/PRESET.Button40AssignedPreset=
pw /MEDIA/PRESET.Button41AssignedPreset=
pw /MEDIA/PRESET.Button42AssignedPreset=
pw /MEDIA/PRESET.Button43AssignedPreset=
pw /MEDIA/PRESET.Button44AssignedPreset=
pw /MEDIA/PRESET.Button45AssignedPreset=
pw /MEDIA/PRESET.Button46AssignedPreset=
pw /MEDIA/PRESET.Button47AssignedPreset=
pw /MEDIA/PRESET.Button48AssignedPreset=
m- /MEDIA/PRESET:createButtonAssignedPreset
m- /MEDIA/PRESET:rename
m- /MEDIA/PRESET:create
m- /MEDIA/PRESET:delete
m- /MEDIA/PRESET:copy
}
6B89#CLOSE /MEDIA/PRESET
6B8A#GET /MEDIA/PRESET/Horizontal.*
{6B89
c- /MEDIA/PRESET
}

**Comment by johnny4x4** [comment]
No it looks like that was everything.

Unfortunately I am not in an environment where I can put this switch on the net.

**Comment by dnmeid** [comment]
OK, so there is an update where I hope you can get the preset names and also recall them.
Can you please create a second preset at the matrix and check if this works. (Presets names are only retrieved from the matrix at module start, not dynamically)
Can you please check if everything else is still working, like crosspoint and feedbacks.

[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11691427/pkg.tgz)

**Comment by dnmeid** [comment]
Let's see if this is better
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11691719/pkg.tgz)

**Comment by dnmeid** [comment]
Ah, I see the responses in the log you posted earlier had been of by one. I have a good feeling about this one now:
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11691887/pkg.tgz)

**Comment by dnmeid** [comment]
Did you delete that button and recreate it? It is normal if we solve problems with options, that it shows the ??() if a wrong value was stored previously. So please delete that button and create it new.

**Comment by johnny4x4** [comment]
It is a new button- I normally dont use the LW presets, I've been using companion to route each source individually. I am only adding the button now to help try and debug.

**Comment by dnmeid** [comment]
Ok, thank you. You see how hard it is to debug some stuff it there is no device available.
Can you go back to the Lightware LDC to the advanced view. there is a checkbox like "Enable editing". If you check it, you'll get a command line entry. Can you please enter the command "GET /MEDIA/PRESET" (without the quotes) and send it. What is the result.

**Comment by johnny4x4** [comment]
I know how difficult it is to troubleshoot this sort of thing with no actual device to work on. I appreciate your time and patience working through this despite the hassle.

Here is the result:
> GET /MEDIA/PRESET
<n- /MEDIA/PRESET/Horizontal
<n- /MEDIA/PRESET/TestPreset

**Comment by dnmeid** [comment]
Ok, we are definitely at the right command. Only question now is why the presets are not parsed correctly.
Here is a version with more verbose logging:
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11692186/pkg.tgz)

**Comment by johnny4x4** [comment]
debug: response from 'GET /MEDIA/PRESET' was: n- /MEDIA/PRESET/Horizontal
n- /MEDIA/PRESET/TestPreset
info: 'GET /MEDIA/PRESET' found 2 presets with the names: Horizontal, TestPreset

**Comment by dnmeid** [comment]
Ok, this part of the code is correct. You still don't see the presets in the action option, correct? Even if you delete that button and recreate it again?

**Comment by johnny4x4** [comment]
Correct, presets are still empty in the actions section.

**Comment by johnny4x4** [comment]
Not sure if this helps but when I select different presets, this is what shows up in the advanced view:

2023-06-08 11:15:04 6BED#GET /MEDIA/PRESET/TestPreset.*
2023-06-08 11:15:04 {6BEC
2023-06-08 11:15:04 }
2023-06-08 11:15:04 {6BED
2023-06-08 11:15:04 pw /MEDIA/PRESET/TestPreset.VideoSource=u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;
2023-06-08 11:15:04 pw /MEDIA/PRESET/TestPreset.VideoDestination=uI1;uI3;uI5;uI7;uI2;uI4;uI6;uI8;uI9;uI10;uI11;uI12;uI13;uI14;uI31;uI32;uI1;uI2;uI3;uI4;uI5;uI6;uI7;uI8;uI9;uI10;uI11;uI12;uI13;uI14;uI15;uI48;uI1;uI1;uI1;uI1;uI1;uI1;uI1;uI38;uI38;uI39;uI43;uI31;uI38;u0;uI31;uI39;
2023-06-08 11:15:04 m- /MEDIA/PRESET/TestPreset:load
2023-06-08 11:15:04 m- /MEDIA/PRESET/TestPreset:addLayer
2023-06-08 11:15:04 m- /MEDIA/PRESET/TestPreset:deleteLayer
2023-06-08 11:15:04 m- /MEDIA/PRESET/TestPreset:save
2023-06-08 11:15:04 }
2023-06-08 11:15:06 6BEE#CLOSE /MEDIA/PRESET/TestPreset
2023-06-08 11:15:06 6BEF#OPEN /MEDIA/PRESET/Horizontal
2023-06-08 11:15:06 {6BEE
2023-06-08 11:15:06 c- /MEDIA/PRESET/TestPreset
2023-06-08 11:15:06 }
2023-06-08 11:15:06 6BF0#GET /MEDIA/PRESET/Horizontal
2023-06-08 11:15:06 {6BEF
2023-06-08 11:15:06 o- /MEDIA/PRESET/Horizontal
2023-06-08 11:15:06 }
2023-06-08 11:15:06 6BF1#GET /MEDIA/PRESET/Horizontal.*
2023-06-08 11:15:06 {6BF0
2023-06-08 11:15:06 }
2023-06-08 11:15:06 {6BF1
2023-06-08 11:15:06 pw /MEDIA/PRESET/Horizontal.VideoSource=u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;u;
2023-06-08 11:15:06 pw /MEDIA/PRESET/Horizontal.VideoDestination=uI1;uI2;uI3;uI4;uI5;uI6;uI7;uI8;uI9;uI10;uI11;uI12;uI13;uI14;uI15;uI16;uI1;uI2;uI3;uI4;uI5;uI6;uI7;uI8;uI9;uI10;uI11;uI12;uI13;uI14;uI15;uI48;uI1;uI1;uI1;uI1;uI1;uI1;uI1;uI31;uI35;uI35;uI36;uI37;uI42;u0;uI35;uI31;
2023-06-08 11:15:06 m- /MEDIA/PRESET/Horizontal:load
2023-06-08 11:15:06 m- /MEDIA/PRESET/Horizontal:addLayer
2023-06-08 11:15:06 m- /MEDIA/PRESET/Horizontal:deleteLayer
2023-06-08 11:15:06 m- /MEDIA/PRESET/Horizontal:save
2023-06-08 11:15:06 }

**Comment by dnmeid** [comment]
Thank you, but this seems now to be a problem somwhere else, the module gets the correct names from the matrix.
Here is a version with even more logging:
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11692380/pkg.tgz)

**Comment by johnny4x4** [comment]
debug: Initializing Recall Preset, choices are: 
info: 'GET /MEDIA/PRESET' found 2 presets with the names: Horizontal, TestPreset
debug: CHOICES_PRESETS are now: [object Object],[object Object]
debug: Initializing Recall Preset, choices are: [object Object],[object Object]

**Comment by dnmeid** [comment]
Everything looks fine, the presets should show up in the option. Are we talking from the same thing?
They should be here:
<img width="839" alt="image" src="https://github.com/bitfocus/companion-module-lightware-lw3/assets/8548888/780c21b0-6e34-4591-ae70-8ad52e87c54d">

And once again: if there was a wrong value stored previously you have to delete that 'Recall Preset' action and add it again to see the correct list. You have to do this every time you try a new version.

**Comment by johnny4x4** [comment]
Yes, we are talking about the same thing:
![image](https://github.com/bitfocus/companion-module-lightware-lw3/assets/42984014/b5c87ae6-e1b2-42dd-9aca-9078a6a0e886)

**Comment by dnmeid** [comment]
Is there anything else if you open the dropdown?

**Comment by johnny4x4** [comment]
No, the dropdown is blank.

**Comment by dnmeid** [comment]
Ok, even more verbose logging: 
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11692576/pkg.tgz)

**Comment by johnny4x4** [comment]
debug: Initializing Recall Preset, choices are: []
info: 'GET /MEDIA/PRESET' found 2 presets with the names: Horizontal, TestPreset
debug: CHOICES_PRESETS are now: [[{"id":"Horizontal","label":"Horizontal"},{"id":"TestPreset","label":"TestPreset"}]]
debug: Initializing Recall Preset, choices are: [[{"id":"Horizontal","label":"Horizontal"},{"id":"TestPreset","label":"TestPreset"}]]

**Comment by dnmeid** [comment]
Great, one little dot adds one square bracket too much. I hope this version does work now:
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11692716/pkg.tgz)

**Comment by johnny4x4** [comment]
Haha how funny, that appears to have fixed the issue.
![image](https://github.com/bitfocus/companion-module-lightware-lw3/assets/42984014/6d88cdb2-4b00-4536-a9b0-729543f60215)

**Comment by dnmeid** [comment]
So does it also recall the presets correctly when you run the action?
And do the other commands and feedbacks still work?

**Comment by johnny4x4** [comment]
The regular switching is working normally. I haven't been using feedbacks or the LW built in presets- let me play around with them a little bit and see how the system responds; then I will let you know.

**Comment by johnny4x4** [comment]
Both "source selected" and  "destination selected" feedbacks are working properly but it seems like the "route" feedback is not. Presets are also working.

**Comment by dnmeid** [comment]
Ok, is there something in the log, giving hints for the error with route feedback?

**Comment by johnny4x4** [comment]
Unfortunately not, when I change the route the only message that shows up in the log is that the route change was successful:
info: XPT Result: mO /MEDIA/XP/VIDEO:switch=OK

**Comment by dnmeid** [comment]
Can you change one crosspoint with LDC and see in the log what the response is?
On my MMX matrix here it looks like
```
> 8649#CALL /MEDIA/VIDEO/XP:onClick(O2,I2)
< {8649
< mO /MEDIA/VIDEO/XP:onClick
< }
< CHG /MEDIA/VIDEO/XP.ModifiableConnectionList=M,-,-,-,-,-,-,-;-,M,-,-,-,-,-,-;-,M,-,-,-,-,-,-;M,-,-,-,-,-,-,-;-,-,-,M,-,-,-,-;M,-,-,-,-,-,-,-;M,-,-,-,-,-,-,-;M,-,-,-,-,-,-,-
< CHG /MEDIA/VIDEO/XP.DestinationConnectionList=I1;I2;I2;I1;I4;I1;I1;I1
< CHG /MEDIA/VIDEO/XP.DestinationPortAutoselect=DP;DP;DP;DP;DP;DP;DP;DP
```

**Comment by johnny4x4** [comment]
When I change one crosspoint, I only see one line appear in advanced view:

CHG /MEDIA/XP/VIDEO.DestinationConnectionStatus=I1;I2;I21;I23;I18;I20;I22;I24;I25;I26;I27;I28;I29;I30;I31;I32;I1;I2;I3;I4;I5;I6;I7;I8;I9;I10;I11;I12;I13;I14;I15;I48;I1;I1;I1;I1;I1;I1;I1;I31;I36;I37;I43;I31;I42;0;I39;I39;

**Comment by dnmeid** [comment]
Did you truncate the line or does it really have a `;` at the end?

**Comment by johnny4x4** [comment]
I did not truncate- there is a `;` at the end anytime I switch around.

**Comment by dnmeid** [comment]
OK, try this:
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11693300/pkg.tgz)

**Comment by johnny4x4** [comment]
I didn't see any changes in the feedback or the logs.

**Comment by dnmeid** [comment]
Sorry, typo, I forgont an I.
Now 
[pkg.tgz](https://github.com/bitfocus/companion-module-lightware-lw3/files/11693791/pkg.tgz)

**Comment by johnny4x4** [comment]
When the module reloaded in companion, this was logged:
`parsing pE /MEDIA/XP/VIDEO.DestinationConnectionList %E002: Not exists`

I didn't notice any other changes.

**Comment by johnny4x4** [comment]
Actually, I was playing around with switch some sources around and noticed it logging in companion similar to the log in the LW advanced view:

```
parsing CHG /MEDIA/XP/VIDEO.DestinationPortGuiStatus=F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;F;2;0;0;F;F;F;F;F;F;F;F;F;F;F;F;F;0;F;F;
parsing CHG /MEDIA/XP/VIDEO.DestinationConnectionStatus=I1;I2;I21;I23;I18;I20;I22;I24;I25;I26;I27;I28;I29;I30;I31;I32;I1;I2;I3;I4;I5;I6;I7;I8;I9;I10;I11;I12;I13;I14;I15;I48;I1;I1;I1;I1;I1;I1;I1;I31;I38;I39;I43;I37;I42;0;I39;I31;
received CHG /MEDIA/XP/VIDEO.DestinationConnectionStatus=I1;I2;I21;I23;I18;I20;I22;I24;I25;I26;I27;I28;I29;I30;I31;I32;I1;I2;I3;I4;I5;I6;I7;I8;I9;I10;I11;I12;I13;I14;I15;I48;I1;I1;I1;I1;I1;I1;I1;I31;I38;I39;I43;I37;I42;0;I39;I31;
info: XPT Result: mO /MEDIA/XP/VIDEO:switch=OK
```

And it looks like feedback might be working now.

**Comment by dnmeid** [comment]
So feedback might be working or is working?
And what about the port names? When you use the 'Switch - Select video input for output' action you should see the real names of the ports if you did set names at the matrix. (module restart required to get name updates)
<img width="770" alt="image" src="https://github.com/bitfocus/companion-module-lightware-lw3/assets/8548888/16261e3a-684b-4f43-9693-cab12479d535">

**Comment by johnny4x4** [comment]
It looks like feedback is working. I just haven't had enough time playing around with it to ensure there isn't some configuration that throws the whole thing off. 

Names are working just fine.

**Comment by johnny4x4** [comment]
After experimenting with it for about an hour, everything is behaving as I would expect. Only one small thing I've noticed- the feedback options in companion use the input/output numbers instead of the names. From an integration standpoint, it would probably be easier to set up a system if the names were shown instead of the numbers.

**Comment by dnmeid** [comment]
Yes and no. The benefit of the numbers is that you can also use ports currently not present in your matrix and if you have a large matrix you can adjust the number faster than than with the dropdown.
Since we now have searchable Dropdowns I also could always write the number in front of the label. Maybe I update this in a later version, but if now everything is working I think it is enough for today.
Can you tell me the model of your matrix, then I put it in the known working list.

**Comment by johnny4x4** [comment]
Yes of course, I completely agree that it is not totally necessary- with smaller matrices it wouldn't really be an issue at all. I definitely like the idea of having numbers with the names. Thank you for spending as much time as you did, I really appreciate it.
I am using an MX2-48x48-HDMI20-A-R

**Comment by dnmeid** [comment]
Ok, I pushed it to the repo. Look for one of the next betas.
Thank you for the remote assistance.
I'm closing this issue now as completed
